lib/orderly.rb

```diff
@@ -17,10 +17,12 @@ module Orderly
     end
 
     def html_for_node(node)
-      if node.is_a?(Capybara::Node::Document)
-        page.body
-      elsif node.native.respond_to?(:inner_html)
-        node.native.inner_html
+      if node.is_a?(String)
+        node
+      elsif node.is_a?(Capybara::Node::Document)
+        html_for_node(node.find(:xpath, '//body'))
+      elsif node.native.respond_to?(:to_html)
+        node.native.to_html
       else
         page.driver.evaluate_script("arguments[0].innerHTML", node.native)
```

Comment:
I think you have to use `outerHTML` here, which is equivalent to `to_html` for nokogiri nodes. Otherwise `innerHTML` for an `<img>` returns nothing.